spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/ExpireSnapshotsSparkAction.java

```diff
@@ -276,12 +277,13 @@ private ExpireSnapshots.Result deleteFiles(Iterator<FileInfo> files) {
     }
 
     LOG.info("Deleted {} total files", summary.totalFilesCount());
-
-    return new BaseExpireSnapshotsActionResult(
-        summary.dataFilesCount(),
-        summary.positionDeleteFilesCount(),
-        summary.equalityDeleteFilesCount(),
-        summary.manifestsCount(),
-        summary.manifestListsCount());
+    return ImmutableResult.builder()
```

Comment:
I am not familiar with this lib but won't the name conflict if we migrate more results to auto-generated beans as it does not take the parent class name into account?

```
import org.apache.iceberg.actions.ImmutableResult;
```

Reply:
We call it `Result` in all actions and use outer class for identification.

Reply:
good catch. I will check if it is possible to alter the generated class name or include the outer class name. 
If I don't find anything in a few hours, I will just use a new private Inner class in `ExpireSnapshotsSparkAction.java`. 

Later It can be easily replaced (as it is private) with `Immutables` once we have better ways in a follow-up PR.
I don't want to block this PR (targeted for 1.2.0).